### ISSUE TITLE
Add an option to configure how boolean fields are exported

### ISSUE THREAD

**Issue by grantfitzsimmons** [specify/specify7#4418]
This is a request to export a different value for yesNo fields:

>Right now you get a 1 or a 0 in the csv which isn't always very helpful, but being able to pick a title value (true/false or yes/no) would be very nice as well

## Current
| Catalog Number | YesNo1 |
|:--------------:|:------:|
|   123456789    |   0    |
|   234567890    |   1    |
|   345678901    |   1    |
|   456789012    |   0    |
|   567890123    |   0    |
|   678901234    |   1    |
|   789012345    |   1    |
|   890123456    |   0    |
|   901234567    |   1    |

## Requested
| Catalog Number | YesNo1 |
|:--------------:|:------:|
|   123456789    | false  |
|   234567890    |  true  |
|   345678901    |  true  |
|   456789012    | false  |
|   567890123    | false  |
|   678901234    |  true  |
|   789012345    |  true  |
|   890123456    | false  |
|   901234567    |  true  |

**Requested By:** Dan B at CSIRO ([Asana](https://app.asana.com/0/0/1203981304440304/1204663198443053/f))

**Comment by maxpatiiuk** [comment]
Duplicate of https://github.com/specify/specify7/issues/2349

**Comment by grantfitzsimmons** [comment]
@maxpatiiuk While I agree it is a duplicate of point #2 in that issue, I think it should be tracked independently. Issues like #2349 make it difficult to prioritize when it's unclear what the exact goal is. I would recommend breaking it into several issues, including this one, rather than closing this as a duplicate.

**Comment by maxpatiiuk** [comment]
My reasoning was:
- Let's keep all discussion regarding improving handling of boolean fields in a single place
- When developer goes to fix this, it would be less context switching to fix all 3 issues at once
- All 3 points mentioned in https://github.com/specify/specify7/issues/2349 were requested by CSIRO, and all are valid requests that we should support

"make it difficult to prioritize when it's unclear what the exact goal is" because the scope is too broad or why?

can you elaborate on your reasoning please, so that I can adjust in the future (it's not the first time I create a "super" issue with several sub items in it)

would you prefer we have a super issue, and then create a sub-issue for each sub item and link them together?

---

feel free to reopen this issue and link ti to #2349 if you disagree with any of the points I mentioned

**Comment by grantfitzsimmons** [comment]
> "make it difficult to prioritize when it's unclear what the exact goal is" because the scope is too broad or why?
> 
> can you elaborate on your reasoning please, so that I can adjust in the future (it's not the first time I create a "super" issue with several sub items in it)

While I agree that the points mentioned in #2349 are valid and that we should support them, I don't agree that making a single large issue is a good idea. If a set of issues/a single complex issue can be separated into multiple issues, that makes them easier to resolve.

For example, here is #2349:

> From CSIRO:
> 
> > 1 - We would like to add an option to be able to choose what values are displayed in Specify query results for yesNo fields for the 3 possible options: NULL/0/1
> > 2 - It would be very useful if we could configure what values are exported out into the csv file - it can be confusing for users to see numeric values when they expect a yes/no or true/false value
> > 3 - Is it possible to set a default value for a yesNo tickbox? When viewing an unticked box on a form it is impossible to know if the database value is NULL or 0 - this can cause users to not find all the relevant data if they don't set up the query correctly. This could be fixed if we could set the default value to 0 for unticked boxes but we don't see a way to do that via the form xml? (Having three possible values for a binary result is a little counter-intuitive!)

If a developer wants to solve this issue, they will now need to:
1. Add an option to choose which value is displayed in the query builder for boolean fields
2. Add the ability to change what is shown in an exported CSV
3. Add the ability to set a default value for boolean fields (this may be possible already, but regardless)

While I understand your concerns about context switching, the larger an issue is the more is required to resolve it, therefore reducing the chance that _this_ group of issues will be solved. Bite sized issues are mcuh easier to tackle, and we can link them together with comments like 'related to #xyz', but forcing them into a single issue makes it difficult to prioritize individual portions of larger issues. We can't put just one of these requests in the milestones because we would need to address all 3 points to close it. It compels us to either separate these or spend additional time resolving a component that may not be worth it compared to other urgent matters. Even though it may take more time later to return to that issue if not done at once, it is better to solve one than none.

A 'super issue' is essentially an unofficial milestone in my view. I think sub-issues for each component would be better so that we can add those components to our milestones and consider them individually. I'll leave this closed for now, but I think we should consider this approach in the future. I'm curious to know @CarolineDenis's opinion on this matter as it mainly concerns her and the development team.

**Comment by maxpatiiuk** [comment]
I agree with your reasoning. Basically I was considering the best case scenario in the ideal world, and you pointed out the practical realities of how we work and prioritize things

Modified https://github.com/specify/specify7/issues/2349#issue-1422503852 to link to this issue

**Comment by CarolineDenis** [comment]
My opinion is very similar to: 

> Bite sized issues are mcuh easier to tackle, and we can link them together with comments like 'related to #xyz', but forcing them into a single issue makes it difficult to prioritize individual portions of larger issues. We can't put just one of these requests in the milestones because we would need to address all 3 points to close it. It compels us to either separate these or spend additional time resolving a component that may not be worth it compared to other urgent matters.

I would keep small issues rather than one super.
